central/splunk/compliance.go

```diff
@@ -27,6 +27,21 @@ var (
 	}
 
 	log = logging.LoggerForModule()
+
+	getClusterIDs = func(ctx context.Context) ([]string, error) {
+		clusterDS := clusterDatastore.Singleton()
+
+		clusters, err := clusterDS.GetClusters(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		clusterIDs := make([]string, len(clusters))
+		for i, cluster := range clusters {
+			clusterIDs[i] = cluster.GetId()
+		}
+		return clusterIDs, nil
+	}
```

Comment:
Originally, this function was named. I moved it into the var block to enable testing.
The problem is that `getClusterIDs` tries to access the `clusterDatastore`, which would require more mocking for tests to pass.
Moving this into a var allows me to override it with a mocked `getClusterIDs` in the tests. This is the function in `compliance_test.go` that [you've asked about a bit later in this review](https://github.com/stackrox/stackrox/pull/3010#discussion_r975010150).
If there is a cleaner way of mocking this, or even a different point where to mock, I'm happy to update this. I picked this spot as it struck the right balance between needed changes and complexity.

Reply:
> Moving this into a var allows me to override it with a mocked getClusterIDs in the tests

So maybe this function should be passed as a parameter? This would allow explicit overrides, as currently it happens behind the scenes and it is difficult to grasp why it is implemented like this. 

If passing this func as a param is an overkill, then we need at least a comment mentioning:
- why we are doing it like this
- where the function `getClusterIDs` is being called

Reply:
I've updated the code to pass a function. To keep the external signature the same, I've introduced a facade that uses the default `getClusterIDs` function. The test uses the mock function and directly calls the internal version of `newComplianceHandler` with the prepared mock. I concur that this is way more readable. 
Tested and verified in a local Docker deployment that the URL path still works as expected and returns results.